server/server.go

```diff
@@ -360,6 +361,7 @@ func (s *Server) ListenAndServe() error {
 // UnixListenAndServe listens on the Unix socket address and handles keyless requests.
 func (s *Server) UnixListenAndServe() error {
```

Comment:
I think instead you should call `defer os.Remove(s.UnixAddr)` and have this occur as part of the cleanup.